### ISSUE TITLE
appAuthRedirectScheme only picked up from app/AndroidManifest.xml

### ISSUE THREAD

**Issue by FreeWillaert** [openid/AppAuth-Android#323]
Applying the readme instructions on a vanilla download, I configured redirect_uri (and other settings) in auth_config.json and set a matching value for appAuthRedirectScheme in the app build.gradle.
When I run this, I get the `redirect_uri is not handled by any activity in this app` error.

Only after changing app/AndroidManifest.xml by removing the example 'https' RedirectUriReceiverActivity and replacing it with

        <activity android:name="net.openid.appauth.RedirectUriReceiverActivity">
            <intent-filter>
                <action android:name="android.intent.action.VIEW"/>
                <category android:name="android.intent.category.DEFAULT"/>
                <category android:name="android.intent.category.BROWSABLE"/>
                <data android:scheme="${appAuthRedirectScheme}"/>
            </intent-filter>
        </activity>

the isRedirectUriRegistered() test succeeds and the error goes away. 

Not sure what's wrong, but it seems to me that the 'merge' of the activity declaration in the app and library manifest is going wrong.
Or am I missing something?

**Comment by psignoret** [comment]
I am experiencing the same issue when running the `debug` build variant. I believe this can be attributed to the following lines from [`library/build.gradle`](https://github.com/openid/AppAuth-Android/blob/master/library/build.gradle#L14):

```gradle
// Define a default value for building the tests. This is not used when building any dependent
// libraries or apps.
android.buildTypes.debug.manifestPlaceholders = [
    'appAuthRedirectScheme': 'net.openid.appauth.test'
];
```

From what I can tell, this will take precedence over any `appAuthRedirectScheme` defined in `defaultConfig` in `app/build.gradle`.

I'm using Android Gradle Plugin 3.1.1 and build tools 27.0.3.

**Comment by iainmcgin** [comment]
@psignoret thanks for point this out - something must have changed in how the build tools handle manifest placeholders, as previously this did not cause any conflict. I'll put in a change ASAP to remove the conflict.

**Comment by iainmcgin** [comment]
This should now be fixed by #341.
